tests/test_utils_metrics.py

```diff
@@ -0,0 +1,116 @@
+#
+#     * Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+#     * Redistributions in binary form must reproduce the above
+# copyright notice, this list of conditions and the following disclaimer
+# in the documentation and/or other materials provided with the
+# distribution.
+#     * Neither the name of Google Inc. nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import absolute_import
+
+import collections
+import platform
+import unittest
+
+import pkg_resources
+
+from google import gax
+from google.gax.utils import metrics
+
+
+# pylint: disable=no-member
+GRPC_VERSION = pkg_resources.get_distribution('grpcio').version
+# pylint: enable=no-member
+
+
+class TestFill(unittest.TestCase):
+    def test_no_argument(self):
+        headers = metrics.fill()
+
+        # Assert that the headers are set appropriately.
+        self.assertEqual(headers['gl-python'], platform.python_version())
+        self.assertEqual(headers['gax'], gax.__version__)
+        self.assertEqual(headers['grpc'], GRPC_VERSION)
+
+        # Assert that the headers are in the correct order.
+        self.assertEqual(
+            [k for k in headers.keys()],
+            ['gl-python', 'gax', 'grpc'],
+        )
+
+    def test_gapic(self):
+        headers = metrics.fill(collections.OrderedDict((
+            ('gl-python', platform.python_version()),
+            ('gapic', '1.0.0'),
+        )))
+
+        # Assert that the headers are set appropriately.
+        self.assertEqual(headers['gl-python'], platform.python_version())
+        self.assertEqual(headers['gapic'], '1.0.0')
+        self.assertEqual(headers['gax'], gax.__version__)
+        self.assertEqual(headers['grpc'], GRPC_VERSION)
+
+        # Assert that the headers are in the correct order.
+        self.assertEqual(
+            [k for k in headers.keys()],
+            ['gl-python', 'gapic', 'gax', 'grpc'],
+        )
+
+
+class TestStringify(unittest.TestCase):
+    def test_no_argument(self):
+        s = metrics.stringify()
```

Comment:
Please don't use a single-character variable name here and below: Google Python style guide permits them only as [counters and iterators](https://google.github.io/styleguide/pyguide.html#Naming).

Something more descriptive (like `empty_metrics`?) would be nice.